apps/papra-server/src/modules/intake-emails/intake-emails.usecases.ts

```diff
@@ -31,13 +45,66 @@ export async function createIntakeEmail({
     intakeEmailsRepository,
   });
 
-  const { emailAddress } = await intakeEmailsServices.generateEmailAddress();
+  let currentUser, currentOrganization;
+  // Check if the lookup for the user and organization hints are needed?
+  if (intakeEmailsServices.name === SCHEMA_USERNAME_INTAKE_EMAIL_DRIVER_NAME) {
+    // userId cannot be null or undefined because of the check "ensureUserIsInOrganization"
+    const { user } = await usersRepository.getUserById({ userId });
 
-  const { intakeEmail } = await intakeEmailsRepository.createIntakeEmail({ organizationId, emailAddress });
+    if (isNil(user)) {
+      // Should not happen, for type-safety
+      throw createError({
+        statusCode: 404,
+        message: 'User not found',
+        code: 'user.not_found',
+      });
+    }
+
+    currentUser = user;
+
+    const { organization } = await organizationsRepository.getOrganizationById({
+      organizationId,
+    });
+
+    if (isNil(organization)) {
+      throw createError({
+        statusCode: 404,
+        message: 'Organization not found',
+        code: 'organization.not_found',
+      });
+    }
+
+    currentOrganization = organization;
+  }
+
+  const { emailAddress } = await intakeEmailsServices.generateEmailAddress(
+    currentUser,
+    currentOrganization,
+  );
+
+  await checkIntakeEmailExists(emailAddress, intakeEmailsRepository);
+
+  const { intakeEmail } = await intakeEmailsRepository.createIntakeEmail({
+    organizationId,
+    emailAddress,
+  });
```

Comment:
The goal of having multiple agnostic drivers implement the same interface is to avoid driver-specific logic in the use case layer. This type of logic should either be implemented consistently for all drivers, or it should be encapsulated within the driver (through dependency injection of repositories for example)
 
Some kind of [strategy pattern](https://refactoring.guru/design-patterns/strategy) with dependency inversion principle